### PR TITLE
Extra fields in transaction and smart contract results structure

### PR DESCRIPTION
**Pull request by miiu96**
- Added extra fields in the `transaction` structure and the `smart contract result` structure

### CODE REVIEW

**Review by bogdan-rosianu — APPROVED** [review]

File: integrationtests/testdata/multiTransferWithScCallAndErrorSignaledBySC/transaction-after-execution-of-scr-dst-shard.json [hunk 1]
```diff
@@ -6,10 +6,15 @@
...
   "signature": "",
   "fee": "1904415000000000",
+  "feeNum": 0.001904415,
   "esdtValues": [
     "4100000000000000000",
     "5482813151746926867746345"
   ],
+  "esdtValuesNum": [
+    4.1,
+    5.482813151746927e+06
```
Comment (iulianpascalau) [inline integrationtests/testdata/multiTransferWithScCallAndErrorSignaledBySC/transaction-after-execution-of-scr-dst-shard.json:16]: This is what I've talked about: discrepancy between `4.1` and `5.48 * 10^-6` 
This is just the way floats work, nothing to do about

File: process/elasticproc/converters/balanceToFloat_test.go [hunk 2]
```diff
@@ -51,6 +51,31 @@ func TestComputeBalanceAsFloat(t *testing.T) {
...
+func TestComputeBalanceToFloat18Decimals(t *testing.T) {
+	t.Parallel()
+
+	ap, _ := NewBalanceConverter(18)
+	require.NotNil(t, ap)
+
+	require.Equal(t, 1e-18, ap.ComputeESDTBalanceAsFloat(big.NewInt(1)))
+	require.Equal(t, 1e-17, ap.ComputeESDTBalanceAsFloat(big.NewInt(10)))
+	require.Equal(t, 1e-16, ap.ComputeESDTBalanceAsFloat(big.NewInt(100)))
+	require.Equal(t, 1e-15, ap.ComputeESDTBalanceAsFloat(big.NewInt(1000)))
```
Comment (iulianpascalau) [inline process/elasticproc/converters/balanceToFloat_test.go:63]: can you add a test with 0?

File: process/elasticproc/converters/balanceToFloat_test.go [hunk 3]
```diff
@@ -51,6 +51,31 @@ func TestComputeBalanceAsFloat(t *testing.T) {
...
+}
+
+func TestComputeBalanceToFloatSliceOfValues(t *testing.T) {
+	t.Parallel()
+
+	ap, _ := NewBalanceConverter(18)
+	require.NotNil(t, ap)
+
+	values := []string{"1000000000000000000", "200000000000000000", "100", "2000"}
```
Comment (iulianpascalau) [inline process/elasticproc/converters/balanceToFloat_test.go:72]: also here? A test with 0?

File: process/elasticproc/converters/balanceToFloat_test.go (hunk 2 above)
Comment (miiu96) [reply process/elasticproc/converters/balanceToFloat_test.go:63]: added

File: process/elasticproc/converters/balanceToFloat_test.go (hunk 3 above)
Comment (miiu96) [reply process/elasticproc/converters/balanceToFloat_test.go:72]: added

File: process/elasticproc/converters/balanceToFloat_test.go [hunk 4]
```diff
@@ -69,8 +70,8 @@ func TestComputeBalanceToFloatSliceOfValues(t *testing.T) {
 	ap, _ := NewBalanceConverter(18)
 	require.NotNil(t, ap)
 
-	values := []string{"1000000000000000000", "200000000000000000", "100", "2000"}
-	require.Equal(t, []float64{1, 0.2, 1e-16, 2e-15}, ap.ComputeSliceOfStringsAsFloat(values))
+	values := []string{"1000000000000000000", "200000000000000000", "100", "2000", "0"}
+	require.Equal(t, []float64{1, 0.2, 1e-16, 2e-15, float64(0)}, ap.ComputeSliceOfStringsAsFloat(values))
```
Comment (bogdan-rosianu) [inline process/elasticproc/converters/balanceToFloat_test.go:74]: could have been `0` instead of `float64(0)`

**Review by iulianpascalau — APPROVED** [review]

File: process/elasticproc/converters/balanceToFloat_test.go (hunk 4 above)
Comment (miiu96) [reply process/elasticproc/converters/balanceToFloat_test.go]: fixed

**Review by bogdan-rosianu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**